server/application-server/src/main/java/de/tum/in/www1/hephaestus/leaderboard/SlackMessageService.java

```diff
@@ -0,0 +1,133 @@
+import org.springframework.scheduling.annotation.EnableScheduling;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Service;
+
+import com.slack.api.bolt.App;
+import com.slack.api.methods.SlackApiException;
+import com.slack.api.methods.request.chat.ChatPostMessageRequest;
+import com.slack.api.methods.response.auth.AuthTestResponse;
+import com.slack.api.methods.response.chat.ChatPostMessageResponse;
+import com.slack.api.model.block.LayoutBlock;
+
+import static com.slack.api.model.block.Blocks.*;
+import static com.slack.api.model.block.composition.BlockCompositions.*;
+import static com.slack.api.model.block.element.BlockElements.*;
+
+@Order(value = Ordered.LOWEST_PRECEDENCE)
+@EnableScheduling
+@Service
+public class SlackMessageService {
+
+    private static final Logger logger = LoggerFactory.getLogger(SlackMessageService.class);
+
+    @Value("${slack.channelId}")
+    private String channelId;
+
+    @Value("${slack.runScheduledMessage}")
+    private boolean runScheduledMessage;
+
+    @Autowired
+    private App slackApp;
+
+    @Autowired
+    private LeaderboardService leaderboardService;
+
+    public void sendMessage(String channelId, List<LayoutBlock> blocks, String fallback)
+            throws IOException, SlackApiException {
+        ChatPostMessageRequest request = ChatPostMessageRequest.builder()
+                .channel(channelId)
+                .blocks(blocks)
+                .text(fallback) // used for example in notifications
+                .build();
+
+        ChatPostMessageResponse response = slackApp.client().chatPostMessage(request);
+
+        if (!response.isOk()) {
+            logger.error("Failed to send message to Slack channel: " + response.getError());
+        }
+    }
+
+    private List<LeaderboardEntryDTO> getTop3SlackReviewers() {
+        LocalDate after = LocalDate.now().minusDays(8);
+        LocalDate before = LocalDate.now().minusDays(1);
+        var leaderboard = leaderboardService.createLeaderboard(Optional.of(after), Optional.of(before),
+                Optional.empty());
+        return leaderboard.subList(0, Math.min(3, leaderboard.size()));
+    }
+
+    @Scheduled(cron = "${slack.cronSchedule}")
+    public void sendScheduledLeaderboard() {
+        if (!runScheduledMessage) {
+            return;
+        }
+
+        var top3reviewers = getTop3SlackReviewers();
+        logger.info("Top 3 Accounts of the last week: " + top3reviewers);
+
+        logger.info("Sending scheduled message to Slack channel...");
+        List<LayoutBlock> blocks = asBlocks(
+                header(header -> header.text(
+                        plainText(pt -> pt.text(":newspaper: Developer Reviews of the last week :newspaper:")))),
+                divider(),
+                context(context -> context
+                        .elements(List.of(markdownText("*October 31, 2024* | Hephaestus Announcement")))),
```

Comment:
I think we can also remove the `Hephaestus Announcement`, we already have the bot named Hephaestus